### ISSUE TITLE
gluonts issue with numpy 1.24+: AttributeError: module 'numpy' has no attribute 'bool'

### ISSUE THREAD

**Issue by ghost** [awslabs/gluon-ts#2509]
## Description
The following error with numpy 1.24 happens on an import 

```
AttributeError: module 'numpy' has no attribute 'bool'
```

## To Reproduce

* install `gluonts`
* install `numpy>=1.24`
* run the following import

```python
from gluonts.mx.trainer import Trainer
```

## Error message or code output

```
Traceback (most recent call last):
  ---------------------------------- redacted from here
    from gluonts.mx.trainer import Trainer
  File "/home/runner/work/pricing-service/pricing-service/.tox/pricing-cfo/lib/python3.9/site-packages/gluonts/mx/__init__.py", line 21, in <module>
    import mxnet as mx
  File "/home/runner/work/pricing-service/pricing-service/.tox/pricing-cfo/lib/python3.9/site-packages/mxnet/__init__.py", line 33, in <module>
    from . import contrib
  File "/home/runner/work/pricing-service/pricing-service/.tox/pricing-cfo/lib/python3.9/site-packages/mxnet/contrib/__init__.py", line 30, in <module>
    from . import text
  File "/home/runner/work/pricing-service/pricing-service/.tox/pricing-cfo/lib/python3.9/site-packages/mxnet/contrib/text/__init__.py", line 23, in <module>
    from . import embedding
  File "/home/runner/work/pricing-service/pricing-service/.tox/pricing-cfo/lib/python3.9/site-packages/mxnet/contrib/text/embedding.py", line 36, in <module>
    from ... import numpy as _mx_np
  File "/home/runner/work/pricing-service/pricing-service/.tox/pricing-cfo/lib/python3.9/site-packages/mxnet/numpy/__init__.py", line 23, in <module>
    from .multiarray import *  # pylint: disable=wildcard-import
  File "/home/runner/work/pricing-service/pricing-service/.tox/pricing-cfo/lib/python3.9/site-packages/mxnet/numpy/multiarray.py", line 47, in <module>
    from .utils import _get_np_op
  File "/home/runner/work/pricing-service/pricing-service/.tox/pricing-cfo/lib/python3.9/site-packages/mxnet/numpy/utils.py", line 37, in <module>
    bool = onp.bool
  File "/home/runner/work/pricing-service/pricing-service/.tox/pricing-cfo/lib/python3.9/site-packages/numpy/__init__.py", line 284, in __getattr__
    raise AttributeError("module {!r} has no attribute "
AttributeError: module 'numpy' has no attribute 'bool'
```

## Environment
- Operating system: ubuntu 
- Python version: 3.9
- GluonTS version: 0.11.5
- MXNet version: 1.9.1

```
Installing collected packages: pytz, urllib3, unittest-xml-reporting, ujson, typing-extensions, tqdm, toolz, threadpoolctl, six, PyYAML, pymysql, protobuf, packaging, nvidia-cuda-runtime-cu11, nvidia-cuda-nvrtc-cu11, nvidia-cublas-cu11, numpy, multidict, MarkupSafe, lightning-utilities, joblib, idna, greenlet, graphviz, fsspec, frozenlist, charset-normalizer, certifi, attrs, async-timeout, yarl, tensorboardX, sqlalchemy, scipy, requests, python-dateutil, pydantic, pyarrow, nvidia-cudnn-cu11, Mako, aiosignal, torch, scikit-learn, pandas, mxnet, alembic, aiohttp, torchtext, torchmetrics, gluonts, pytorch_lightning
Successfully installed Mako-1.2.4 MarkupSafe-2.1.1 PyYAML-6.0 aiohttp-3.8.3 aiosignal-1.3.1 alembic-1.9.0 async-timeout-4.0.2 attrs-22.1.0 certifi-2022.12.7 charset-normalizer-2.1.1 frozenlist-1.3.3 fsspec-2022.11.0 gluonts-0.11.5 graphviz-0.8.4 greenlet-2.0.1 idna-3.4 joblib-1.2.0 lightning-utilities-0.4.2 multidict-6.0.3 mxnet-1.9.1 numpy-1.24.0 nvidia-cublas-cu11-11.10.3.66 nvidia-cuda-nvrtc-cu11-11.7.99 nvidia-cuda-runtime-cu11-11.7.99 nvidia-cudnn-cu11-8.5.0.96 packaging-22.0 pandas-1.5.2 protobuf-3.20.1 pyarrow-10.0.1 pydantic-1.10.2 pymysql-1.0.2 python-dateutil-2.8.2 pytorch_lightning-1.8.5.post0 pytz-2022.7 requests-2.28.1 scikit-learn-1.2.0 scipy-1.9.3 six-1.16.0 sqlalchemy-1.4.45 tensorboardX-2.5.1 threadpoolctl-3.1.0 toolz-0.12.0 torch-1.13.1 torchmetrics-0.11.0 torchtext-0.14.1 tqdm-4.64.1 typing-extensions-4.4.0 ujson-5.6.0 unittest-xml-reporting-3.0.4 urllib3-1.26.13 yarl-1.8.2
```

**Comment by lostella** [comment]
Thanks @raffienficiaud-reasonal, we're on it, see #2506 :-)

**Comment by jaheba** [comment]
Just some context: The problem is with `mxnet` since it's using `np.bool` -- which was removed in the latest `numpy` release.

We could restrict the numpy version to `<=1.23`, but that doesn't feel right. So for now, we just limit the version if one uses `pip install gluonts[mxnet]`.

**Comment by lostella** [comment]
Closing this as v0.11.6 was released, which forces `numpy<1.24` when doing `pip install gluonts[mxnet]`.

In case anyone runs into this issue, please consider downgrading NumPy by doing

```
pip install numpy<1.24
```
